components/server/src/code-sync/code-sync-service.ts

```diff
@@ -303,4 +289,27 @@ export class CodeSyncService {
         return router;
     }
 
+    protected async deleteObjects(userId: string, resourceKey: ServerResource, revs: string[]) {
+        const tasks = revs.map(rev => this.db.deleteResource(userId, resourceKey, rev, async (rev: string) => {
+            const obj = toObjectName(resourceKey, rev);
+            try {
+                const request = new DeleteRequest();
+                request.setOwnerId(userId);
+                request.setExact(obj);
+                const blobsClient = this.blobsProvider.getDefault();
+                await util.promisify<DeleteRequest, DeleteResponse>(blobsClient.delete.bind(blobsClient))(request);
+            } catch (err) {
+                if (err.code === status.NOT_FOUND) {
+                    return;
+                }
+                log.error({ userId }, 'code sync: failed to delete obj', err, { object: obj });
+                throw err;
+            }
+        }).catch(err => {
+            log.error({ userId }, 'code sync: failed to delete', err);
+        }))
+        await Promise.allSettled(tasks)
```

Comment:
missing semicolon, no need to await the last promise just return it

Reply:
await first and return void to avoid caller use this to check result 😄 https://github.com/gitpod-io/gitpod/pull/8644#discussion_r822938267